app/workers/rapid_ready_for_decision/form526_hypertension_job.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'lighthouse/veterans_health/client'
+
+module RapidReadyForDecision
+  class Form526HypertensionJob < Form526BaseJob
+    STATSD_KEY_PREFIX = 'worker.fast_track.form526_hypertension_job'
+
+    sidekiq_options retry: 2
+
+    def assess_data(form526_submission)
+      assessed_data = query_and_assess_lighthouse(form526_submission)
+
+      return if assessed_data[:bp_readings].blank?
+
+      assessed_data
+    end
+
+    private
+
+    def query_and_assess_lighthouse(form526_submission)
+      client = lighthouse_client(form526_submission)
+      bp_readings = assess_bp_readings(client.list_resource('observations'))
+      # stop querying if bp_readings.blank?
+      medications = assess_medications(client.list_resource('medications')) if bp_readings.present?
+
+      {
+        bp_readings: bp_readings,
+        medications: medications
+      }
+    end
+
+    def assess_bp_readings(bp_readings)
+      return RapidReadyForDecision::HypertensionObservationData.new(bp_readings).transform if bp_readings.present?
```

Comment:
This is only my personal preference, but because this line is pretty long on the eyes, and contains most of the moving parts, I find it more readable to split it up, short-circuit style

```ruby
return [] unless bp_readings.present?

RapidReadyForDecision::HypertensionObservationData.new(bp_readings).transform
```

YMMV and I'm fine with the code in its current form too.

Reply:
That's much better! Thanks!